### DISCUSSION TITLE
Navigating through the documentation

### DISCUSSION THREAD

**Discussion by nitishfy** [checkstyle/checkstyle#12303] [Q&A] [answered]
Hey folks, 

I'd like to take a basic suggestion (which can prove to be helpful for newbies) :

Since the documentation is large, helpful and overwhelming at the same time, Is it recommended to go through the documentation as per the slides made? For eg. Configuration, Running.... 
_Or_
Should a newbie first of all try to run(use) checkstyle and then go through the documentation topics one by one?

**Comment by romani** [comment]
Yes, that is why first issues to fix we recommend on extension of documentation, where user learn how to use us by CLI.
Usually users use is by plugins in building system or IDEs.

**Comment by nitishfy** [comment]
@romani So first use checkstyle and then go through documentation?

**Comment by romani** [comment, marked as answer]
Yes, ideal case is to start to use it, and understand that checkstyle is configuration based and learn how to control checkstyle by changing configuration, here user will need documentation and know how to find required information in it.
